test/TestUtilities/ApproxArray.hpp

```diff
@@ -0,0 +1,72 @@
+#ifndef SFML_TESTUTILITIES_APPROXARRAY_HPP
+#define SFML_TESTUTILITIES_APPROXARRAY_HPP
+
+#include <doctest.h>
+#include <array>
+
+namespace std
+{
+    template <typename T, std::size_t size>
+    std::ostream& operator<<(std::ostream& out, const std::array<T, size>& array)
+    {
+        if (size == 0)
+            return out << "{}";
+
+        out << "{ ";
+        for (size_t i = 0; i + 1 < array.size(); ++i)
+            out << array[i] << ", ";
+        out << array.back() << " }";
+
+        return out;
+    }
+}
+
+namespace sf::Testing
+{
+    template <typename T, std::size_t size>
+    class ApproxArray
+    {
+    public:
+        ApproxArray(const std::array<T, size>& values)
+        : values{values}
+        {}
+
+        const std::array<T, size>& values;
+    };
```

Comment:
Should this just be a struct so we can depend on the default `public` access of everything?